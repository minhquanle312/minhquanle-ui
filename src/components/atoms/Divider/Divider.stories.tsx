/* eslint-disable no-console */
// Libraries
import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

// Components
import { Divider } from './Divider';

export default {
  title: 'Atoms/Divider',
  component: Divider,
  argTypes: {
    children: {
      name: 'children',
      defaultValue: undefined,
      description: 'The wrapped title',
      table: {
        type: { summary: 'ReactNode' },
        defaultValue: { summary: '-' },
      },
      control: {
        type: null,
      },
    },
  },
  parameters: {
    docs: {
      description: {
        component: 'A divider line separates different content.',
      },
    },
  },
} as ComponentMeta<typeof Divider>;

// Variables

// Default
const Template: ComponentStory<typeof Divider> = args => <Divider {...args} />;

export const Default = Template.bind({});

Default.args = {};

export const Horizontal: ComponentStory<any> = () => (
  <>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista
      probare, quae sunt a te dicta? Refert tamen, quo modo.
    </p>
    <Divider />
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista
      probare, quae sunt a te dicta? Refert tamen, quo modo.
    </p>
    <Divider dashed />
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista
      probare, quae sunt a te dicta? Refert tamen, quo modo.
    </p>
  </>
);

Horizontal.parameters = {
  docs: {
    description: {
      story: 'Divider is horizontal by default. You can add text within Divider.',
    },
  },
};

export const TextWithoutHeadingStyle: ComponentStory<any> = () => (
  <>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista
      probare, quae sunt a te dicta? Refert tamen, quo modo.
    </p>
    <Divider plain>Text</Divider>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista
      probare, quae sunt a te dicta? Refert tamen, quo modo.
    </p>
    <Divider orientation="left" plain>
      Left Text
    </Divider>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista
      probare, quae sunt a te dicta? Refert tamen, quo modo.
    </p>
    <Divider orientation="right" plain>
      Right Text
    </Divider>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista
      probare, quae sunt a te dicta? Refert tamen, quo modo.
    </p>
  </>
);

TextWithoutHeadingStyle.parameters = {
  docs: {
    description: {
      story: 'You can use non-heading style of divider text by setting plain.',
    },
  },
};

export const Vertical: ComponentStory<any> = () => (
  <>
    Text Vertical
    <Divider type="vertical" />
    <a href="#">Link 1 123</a>
    <Divider type="vertical" />
    <a href="#">Link 2 234312</a>
  </>
);

Vertical.parameters = {
  docs: {
    description: {
      story: 'Use `type="vertical"` make it vertical.',
    },
  },
};
