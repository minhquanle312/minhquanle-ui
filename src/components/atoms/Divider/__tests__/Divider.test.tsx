// Libraries
import React from 'react';
import { render } from '@testing-library/react';

// Components
import { Divider } from '../index';
import { Horizontal, TextWithoutHeadingStyle, Vertical, Default } from '../Divider.stories';

describe('Divider Component', () => {
  it('should render the Divider Component', () => {
    const { container } = render(<Divider />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render the Default Divider Component', () => {
    const { container } = render(<Default />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render the Horizontal Divider Component', () => {
    const { container } = render(<Horizontal />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render the Vertical Divider Component', () => {
    const { container } = render(<Vertical />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render the TextWithoutHeadingStyle Divider Component', () => {
    const { container } = render(<TextWithoutHeadingStyle />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
