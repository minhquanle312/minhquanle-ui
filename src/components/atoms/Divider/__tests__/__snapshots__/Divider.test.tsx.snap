// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component should render the Default Divider Component 1`] = `
<div>
  <div
    class="ant-divider css-dev-only-do-not-override-htwhyh ant-divider-horizontal"
    role="separator"
  />
</div>
`;

exports[`Divider Component should render the Divider Component 1`] = `
<div>
  <div
    class="ant-divider css-dev-only-do-not-override-htwhyh ant-divider-horizontal"
    role="separator"
  />
</div>
`;

exports[`Divider Component should render the Horizontal Divider Component 1`] = `
<div>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista probare, quae sunt a te dicta? Refert tamen, quo modo.
  </p>
  <div
    class="ant-divider css-dev-only-do-not-override-htwhyh ant-divider-horizontal"
    role="separator"
  />
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista probare, quae sunt a te dicta? Refert tamen, quo modo.
  </p>
  <div
    class="ant-divider css-dev-only-do-not-override-htwhyh ant-divider-horizontal ant-divider-dashed"
    role="separator"
  />
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista probare, quae sunt a te dicta? Refert tamen, quo modo.
  </p>
</div>
`;

exports[`Divider Component should render the TextWithoutHeadingStyle Divider Component 1`] = `
<div>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista probare, quae sunt a te dicta? Refert tamen, quo modo.
  </p>
  <div
    class="ant-divider css-dev-only-do-not-override-htwhyh ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center ant-divider-plain"
    role="separator"
  >
    <span
      class="ant-divider-inner-text"
    >
      Text
    </span>
  </div>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista probare, quae sunt a te dicta? Refert tamen, quo modo.
  </p>
  <div
    class="ant-divider css-dev-only-do-not-override-htwhyh ant-divider-horizontal ant-divider-with-text ant-divider-with-text-left ant-divider-plain"
    role="separator"
  >
    <span
      class="ant-divider-inner-text"
    >
      Left Text
    </span>
  </div>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista probare, quae sunt a te dicta? Refert tamen, quo modo.
  </p>
  <div
    class="ant-divider css-dev-only-do-not-override-htwhyh ant-divider-horizontal ant-divider-with-text ant-divider-with-text-right ant-divider-plain"
    role="separator"
  >
    <span
      class="ant-divider-inner-text"
    >
      Right Text
    </span>
  </div>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed nonne merninisti licere mihi ista probare, quae sunt a te dicta? Refert tamen, quo modo.
  </p>
</div>
`;

exports[`Divider Component should render the Vertical Divider Component 1`] = `
<div>
  Text Vertical
  <div
    class="ant-divider css-dev-only-do-not-override-htwhyh ant-divider-vertical"
    role="separator"
  />
  <a
    href="#"
  >
    Link 1 123
  </a>
  <div
    class="ant-divider css-dev-only-do-not-override-htwhyh ant-divider-vertical"
    role="separator"
  />
  <a
    href="#"
  >
    Link 2 234312
  </a>
</div>
`;
